README.md

```diff
@@ -0,0 +1,112 @@
+
+## AWS Agents can do better
+If your agents are already running on AWS Instances there is a way better and best practise solution solving your problem: Instance Profiles. Instance profiles allow you to assign Roles to Instances. The AWS SDK will realise running on an AWS Instance and try to retrieve credentials from the AWS Metadata API. The huge benefit of this is you do not need to rotate credentials by yourself as AWS is taking care of it. These credentials will have a short live time so even if they get leaked a user will only have a certain amount of time to use those credentials.
+
+## Not every task should have this privilege
+In DC/OS multiple applications will share the same agent and therefore share the same instance profile. This is something you might want to avoid. With the initially described process you only hand out the users credentials to the applications you’ve selected, so you decide based on the secret containing the credentials which application gets the creds.
+
+## AssumeRole and external_id
+We can combine the security of instance profiles with the selective authorization of DC/OS secrets. AWS offers a process called AssumeRole. With this a Role (Instance,User) is able to retrieve temporary credentials for another role ( even other accounts are possible ).
+So in our example the Instance would Assume into the Role having access to the S3 bucket.
+This process alone does not really change the authorization problem as still every application would be able to use it but AWS gives an additional layer of security to this procedure: external_id. The external_id is a pre-shared-key added to the trust relationship of a role which allows us to assume into it.
+This PSK will allow us to use DC/OS Secrets acting as authorization instance for our application by placing an AssumeRole configuration including the external_id.
+
+## Example
+[This repository](https://github.com/fatz/dcos-secure-instance-profiles) contains a [main.tf](./main.tf) with a example setup. You only need to place a DC/OS license in your Homefolder ( `$HOME/license.txt` ) and the public key file of the SSH-Key you've loaded into your ssh-agent at `~/.ssh/id_rsa.pub`. If these files are at different locations just edit the `main.tf` and change the path for your environment.
+
+### Creating the cluster
+Once you've downloaded all the files of this repository ( `git clone https://github.com/fatz/dcos-secure-instance-profiles && cd dcos-secure-instance-profiles` ) you need to initialize terraform and start creating the cluster
+
+Before you start creating the cluster make sure your AWS setup is finished and working. Either `$AWS_PROFILE` needs to be set to the profile you want to use or make sure you've properly setup your aws cli `aws configure`. To ensure you will use the expected account you can run `aws sts get-caller-identity` and see the account id you will be using.
```

Comment:
Before you start creating the cluster make sure that your AWS setup is finished and working. Either `$AWS_PROFILE` needs to be set to the profile you want to use or make sure that you've properly setup your aws cli `aws configure`. To ensure you will are using the correct account you should run `aws sts get-caller-identity` and see the account id you that will be using.